src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs

```diff
@@ -6251,5 +6251,341 @@ .maxstack  2
+                CompileAndVerify(comp, expectedOutput: expectedResult);
+            }
+
+            // PROTOTYPE: Should the ERR_CheckedOverflow cases be evaluated at runtime rather
+            // than compile time to allow operations to succeed on 64-bit platforms?
+            void constantExpression(string opType, string expr, string expectedResult, DiagnosticDescription[] expectedDiagnostics)
+            {
+                string source =
+$@"using System;
+class Program
+{{
+    static void Main()
+    {{
+        object result;
+        try
+        {{
+            {opType} value = {expr};
+            result = value;
+        }}
+        catch (Exception e)
+        {{
+            result = e.GetType().FullName;
+        }}
+        Console.WriteLine(result);
+    }}
+}}";
+                var comp = CreateCompilation(source, options: TestOptions.ReleaseExe, parseOptions: TestOptions.RegularPreview);
+                comp.VerifyDiagnostics(expectedDiagnostics);
+
+                if (expectedDiagnostics.Length > 0) return;
+
+                CompileAndVerify(comp, expectedOutput: expectedResult);
+            }
+        }
+
+        // OverflowException behavior is consistent with unchecked int division.
+        [Fact]
+        public void UncheckedIntegerDivision()
+        {
+            string source =
+@"using System;
+class Program
+{
+    static void Main()
+    {
+        Console.WriteLine(Execute(() => IntDivision(int.MinValue + 1, -1)));
+        Console.WriteLine(Execute(() => IntDivision(int.MinValue, -1)));
+        Console.WriteLine(Execute(() => IntRemainder(int.MinValue + 1, -1)));
+        Console.WriteLine(Execute(() => IntRemainder(int.MinValue, -1)));
+        Console.WriteLine(Execute(() => NativeIntDivision(int.MinValue + 1, -1)));
+        Console.WriteLine(Execute(() => NativeIntDivision(int.MinValue, -1)));
+        Console.WriteLine(Execute(() => NativeIntRemainder(int.MinValue + 1, -1)));
+        Console.WriteLine(Execute(() => NativeIntRemainder(int.MinValue, -1)));
+    }
+    static object Execute(Func<object> f)
+    {
+        try
+        {
+            return f();
+        }
+        catch (Exception e)
+        {
+            return e.GetType().FullName;
+        }
+    }
+    static int IntDivision(int x, int y) => unchecked(x / y);
+    static int IntRemainder(int x, int y) => unchecked(x % y);
+    static nint NativeIntDivision(nint x, nint y) => unchecked(x / y);
+    static nint NativeIntRemainder(nint x, nint y) => unchecked(x % y);
+}";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseExe, parseOptions: TestOptions.RegularPreview);
+            CompileAndVerify(comp, expectedOutput:
+$@"2147483647
+System.OverflowException
+0
+System.OverflowException
+2147483647
+{(IntPtr.Size == 4 ? "System.OverflowException" : "2147483648")}
+0
+{(IntPtr.Size == 4 ? "System.OverflowException" : "0")}");
+        }
     }
```

Comment:
>} [](start = 4, length = 1)

Unless we already have them, it will probably be useful to have tests that target ExecutionArchitecture.x64 and verify runtime behavior for unchecked/never-checked operators for scenarios when folding cannot be performed at compile time.

Reply:
Tests are executed with x86 and x64.

---
In reply to: [389935658](https://github.com/dotnet/roslyn/pull/42143#discussion_r389935658) [](ancestors = 389935658)

Reply:
> Tests are executed with x86 and x64. 

And we have ability to detect the platform and test accordingly. See NestedIfStatements unit-test, for example.  

---
In reply to: [389936772](https://github.com/dotnet/roslyn/pull/42143#discussion_r389936772) [](ancestors = 389936772,389935658)

Reply:
Perhaps I misunderstood the response. Are you saying we already have tests like that?

---
In reply to: [389941527](https://github.com/dotnet/roslyn/pull/42143#discussion_r389941527) [](ancestors = 389941527,389936772,389935658)

Reply:
The test is detecting whether the platform is 32-bit or 64-bit, using `IntPtr.Size`.

---
In reply to: [389941527](https://github.com/dotnet/roslyn/pull/42143#discussion_r389941527) [](ancestors = 389941527,389936772,389935658)

Reply:
There are a number of tests that test differences in behavior for 32-bit and 64-bit. See use of `IntPtr.Size` in this file.

---
In reply to: [389982721](https://github.com/dotnet/roslyn/pull/42143#discussion_r389982721) [](ancestors = 389982721,389941527,389936772,389935658)